frontend/src/components/GoogleMapLoader.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div>
+    <div class="google-map" ref="googleMap"></div>
+    <template v-if="Boolean(map)">
+      <slot :map="map" />
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+/* eslint-disable no-undef */
+import { defineComponent, PropType } from "vue";
+import { Loader } from "@googlemaps/js-api-loader";
+
+export default defineComponent({
+  props: {
+    mapConfig: {
+      type: Object as PropType<google.maps.MapOptions>,
+      required: true,
+    },
+    apiKey: {
+      type: String,
+      default: "",
+    },
+    markersOptions: {
+      type: Array as PropType<google.maps.MarkerOptions[]>,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      map: null as google.maps.Map | null,
+    };
+  },
+
+  async mounted() {
+    await this.initializeMap();
+  },
+
+  methods: {
+    async initializeMap() {
+      const mapContainer = this.$refs.googleMap as Element;
+      const api = await new Loader({
+        apiKey: this.apiKey,
+      }).load();
+
+      this.map = new api.maps.Map(mapContainer, this.mapConfig);
```

Comment:
nit: maybe possible to disable some of the buttons we dunnid in the map? (e.g Map/Satellite, the small yellow man for the street view)

Reply:
yep, sure, removed! I've also removed the zoom in/out button since I don't think they add much value (and only serve to clutter the UI)